lib/dry/system/container.rb

```diff
@@ -75,7 +73,7 @@ class Container
       setting :name
       setting :default_namespace
       setting(:root, Pathname.pwd.freeze) { |path| Pathname(path) }
-      setting :system_dir, "system"
+      setting :bootable_dirs, ["system/boot"]
```

Comment:
@timriley so, this seems to be a rather serious API breakage. The system dir was meant to be used as a place for keeping core system files, like container definition and auto-inject module. I'm not sure why removing this feature is needed for the bootable refactor. Could this at least be separated in another PR? I'm not against removing it btw, I just need to understand why we're a) removing it here b) hardcoding `system/boot` whereas previously it could be configured

Reply:
@solnic yeah, you’re right, I should pare this PR back to supporting multiple boot dirs only. 

I do think that the system dir setting _could_ be removed before 1.0 but we can definitely have a separate conversation about that.

Reply:
@timriley sounds good 🙂